test/quantization/test_quantized_op.py

```diff
@@ -323,25 +337,30 @@ def test_leaky_relu_observed_output(self, X):
         self._test_activation_function(X, 'leaky_relu', leaky_relu_test_configs)
 
     """Tests the correctness of the quantized::relu op."""
-    @given(X=hu.tensor(shapes=hu.array_shapes(1, 5, 1, 5),
-                       qparams=hu.qparams()),
-           alpha=st.floats(0.0, 1.0, allow_nan=False, allow_infinity=False))
-    def test_leaky_relu(self, X, alpha):
-        X, (scale, zero_point, torch_type) = X
+    def test_leaky_relu(self):
+        shapes = ((4,), (4, 4), (4, 4, 4), (4, 4, 4, 4))
+        dtypes = (torch.quint8, torch.qint8)
+        memory_formats = (torch.channels_last, torch.contiguous_format)
+        test_cases = itertools.product(shapes, dtypes, memory_formats)
+        for shape, dtype, memory_format in test_cases:
+            if memory_format == torch.channels_last and len(shape) != 4:
+                continue
+            X, scale, zero_point, torch_type, alpha = \
+                torch.randn(*shape), 0.1, 0, dtype, 0.01
+            X = X.to(memory_format=memory_format)
 
-        X = torch.from_numpy(X)
-        qX = torch.quantize_per_tensor(X, scale=scale, zero_point=zero_point,
-                                       dtype=torch_type)
-        dqX = qX.dequantize()
+            qX = torch.quantize_per_tensor(X, scale=scale, zero_point=zero_point,
+                                           dtype=torch_type)
+            dqX = qX.dequantize()
 
-        # torch.nn.functional
-        op = torch.nn.functional.leaky_relu
-        dqY = op(dqX, negative_slope=alpha)
-        qY = torch.quantize_per_tensor(dqY, scale=scale, zero_point=zero_point,
-                                       dtype=torch_type)
-        qY_hat = op(qX, negative_slope=alpha)
-        self.assertEqual(qY.dequantize(), qY_hat.dequantize(),
-                         msg="F.leaky_relu failed ({} vs {})".format(qY, qY_hat))
+            # torch.nn.functional
+            op = torch.nn.functional.leaky_relu
+            dqY = op(dqX, negative_slope=alpha)
+            qY = torch.quantize_per_tensor(dqY, scale=scale, zero_point=zero_point,
+                                           dtype=torch_type)
+            qY_hat = op(qX, negative_slope=alpha)
+            self.assertEqual(qY.dequantize(), qY_hat.dequantize(),
+                             msg="F.leaky_relu failed ({} vs {})".format(qY, qY_hat))
```

Comment:
For `leaky_relu` and the ones below, should we be using `_test_activation_function` to be consistent?

Reply:
I think that would be valuable for the future.  We are hoping to land this urgently (FB teams blocked by this, and also to be able to pick into OSS v1.7), so maybe we can improve the test infra as a follow-up PR.  I would also recommend reducing our exposure to hypothesis in these kinds of tests, it seems to be a net negative in my experience so far, as the tensor values it gives are not representative of real inputs, which could mask issues.